File: backend/parking-domain/src/main/java/com/chatchatabc/parking/domain/repository/ParkingRoleRepository.java
package com.chatchatabc.parking.domain.repository;

import com.chatchatabc.parking.domain.model.ParkingRole;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface ParkingRoleRepository extends JpaRepository<ParkingRole, String> {

    /**
     * Find role by name
     *
     * @param name the name
     * @return the role
     */
    Optional<ParkingRole> findByName(String name);
}

File: backend/src/main/kotlin/com/chatchatabc/parking/domain/repository/ParkingRoleRepository.kt
package com.chatchatabc.parking.domain.repository

import com.chatchatabc.parking.domain.model.ParkingRole
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface ParkingRoleRepository : JpaRepository<ParkingRole, String> {
    /**
     * Find role by name
     */
    fun findByName(name: String): Optional<ParkingRole>
}